pkg/builder/builder.go

```diff
@@ -34,7 +32,8 @@ const (
 
 var (
 	defaultRenderConfig = renderConfig{
-		size:        1200,
+		size:        4000,
+		mediumSize:  2000,
```

Comment:
Lots of lines changed in here, but it just adds support for rendering a medium sized image and, importantly, cleans up the rendering pipeline to be way easier to understand and debug. It works, don't worry about the details.